### DISCUSSION TITLE
choose outer face with combinatorial_embedding_to_pos

### DISCUSSION THREAD

**Discussion by ChristopheVuillot** [networkx/networkx#4699] [Q&A] [answered]
Hello,

I'm working with some planar graphs which are triangulated by construction (every face is a triangle even the outer face). I would like to draw the graph using the function combinatorial_embedding_to_pos and I would like to specify which one of the faces should be the outer face for the drawing. Is there already a way to do this ?

I see in the code of combinatorial_embedding_to_pos that I would need to shortcut the triangulate_embedding function and directly give my graph and my chosen outer face.
Maybe there is another way, or could this ability be added ?

Thanks for your help !

**Comment by dschult** [comment, marked as answer]
It looks to me like you want to be able to input the `embedding` and `outer_face` instead of having the function compute them. This seems very reasonable to me. To implement that feature to NetworkX I propose that we make the `triangulate_embedding` function public by including it in `__all__`. Then construct a new function that does all but the top 6-8 lines of `combinatorial_embedding_to_pos` with input `embedding, outer_face`.  The original docstring and function will now need to run it's 6-8 lines and then call the new function. I think the new function should be added to `__all__` as well.

Does this make sense with your understanding as well?

**Reply by ChristopheVuillot** [reply]
Yes, I think that makes sense in my case :)

**Reply by ChristopheVuillot** [reply]
I created a PR doing this: https://github.com/networkx/networkx/pull/4706#issue-602476221
